### DISCUSSION TITLE
Am I allowed to collect the issues/comments/contributions code from other users to make into a dataset?

### DISCUSSION THREAD

**Discussion by huskydoge** [community/community#135466] [Other Feature Feedback, Questions, & Ideas] [answered]
I'm currently working on a research paper that involves collecting issues, comments, and code contributions from public repositories made by other users on GitHub. Although there have already been some relevant works ([The Stack](https://huggingface.co/datasets/bigcode/the-stack/discussions/43), [SWE-Bench](https://arxiv.org/abs/2310.06770)), I'm still worried about potential legal implications, which has been kindly pointed out by my paper reviewers. Therefore, I've carefully reviewed GitHub's Terms of Service. However, I've found the relevant content to be somewhat ambiguous.

The most pertinent section appears to be D.3, paragraph 2 of the [GitHub Terms of Service](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#c-acceptable-use), which states:

> "Because you retain ownership of and responsibility for Your Content, we need you to grant us — and other GitHub Users — certain legal permissions, listed in Sections D.4 — D.7."

Furthermore, Section D.5, ["License Grant to Other Users"](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#5-license-grant-to-other-users), suggests that any User-Generated Content posted publicly, including issues, comments, and contributions to other users' repositories, may be viewed by others. This applies in cases where no specific license is provided.

Despite these clauses, I find myself uncertain about the precise ways in which this content is permitted to be used. The terms don't clearly delineate which uses are allowed and which are prohibited. This lack of clarity has left me somewhat confused about the proper protocol for incorporating such content into my research.

Given these ambiguities, I'm seeking further clarification to ensure my research practices align with both ethical standards and legal requirements. Any insights or guidance you could provide on this matter would be greatly appreciated.

**Comment by TheGiraffe3** [comment, marked as answer]
Because you aren't selling the paper, and if all of the comments are in public repositories, then there won't be legal implications (though do avoid taking comments that are too revealing about people).
